packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -967,21 +967,35 @@ export async function showAttributes(node: api.IZoweDatasetTreeNode, datasetProv
 }
 
 /**
- * Submit the contents of the editor as JCL.
+ * Submit the contents of the editor or file as JCL.
  *
  * @export
  * @param {DatasetTree} datasetProvider - our DatasetTree object
  */
 // This function does not appear to currently be made available in the UI
-export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetTreeNode>): Promise<void> {
+export async function submitJcl(datasetProvider: api.IZoweTree<api.IZoweDatasetTreeNode>, file?: vscode.Uri): Promise<void> {
     ZoweLogger.trace("dataset.actions.submitJcl called.");
-    if (!vscode.window.activeTextEditor) {
-        const errorMsg = localize("submitJcl.noDocumentOpen", "No editor with a document that could be submitted as JCL is currently open.");
-        api.Gui.errorMessage(errorMsg);
-        ZoweLogger.error(errorMsg);
+    if (!vscode.window.activeTextEditor && !file) {
+        const notActiveEditorMsg = localize(
+            "submitJcl.notActiveEditorMsg",
+            "No editor with a document that could be submitted as JCL is currently open."
+        );
+        api.Gui.errorMessage(notActiveEditorMsg);
+        ZoweLogger.error(notActiveEditorMsg);
         return;
     }
+    if (file) {
+        await vscode.commands.executeCommand("filesExplorer.openFilePreserveFocus", file);
+    }
     const doc = vscode.window.activeTextEditor.document;
+    // run check for 'jcl' to be included in the vs code document languageId
+    const jclCheckRegex = /^.*jcl.*/i;
+    if (!jclCheckRegex.test(doc.languageId)) {
```

Comment:
addressed in commit https://github.com/zowe/vscode-extension-for-zowe/pull/2510/commits/097391ccf19c50e96669fdd0006a9d0dc79ed2c2 
Since not checking the file language I updated the displayed text of command to `Submit as JCL` since we aren't checking anymore.
<img width="259" alt="Screen Shot 2023-10-24 at 9 25 50 AM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/49491949/ec81876e-fb1f-47b8-96f2-0f6e29a11270">